File: youqugou-interface/src/main/java/cn/itcast/core/service/SpecificationService.java
package cn.itcast.core.service;


import cn.itcast.core.entity.PageResult;
import cn.itcast.core.pojo.specification.Specification;
import cn.itcast.core.pojogroup.SpecificationVo;


import java.util.List;
import java.util.Map;

public interface SpecificationService {
    PageResult searchSpecification(Integer pageNum, Integer pageSize, Specification specification);

    SpecificationVo selectSpecificationVoById(Long id);

    int insertSpecificationVo(SpecificationVo specificationVo);

    int updateSpecificationVo(SpecificationVo specificationVo);

    int deleteSpecificationVo(Long[] ids);

    List<Map> selectAllSpecificationMap();
}

File: youqugou-web-shop/src/main/webapp/js/service/specificationService.js
app.service("specificationService", function ($http) {
    this.searchSpecification = function (currentPage, itemsPerPage, searchEntity) {
        return $http.post("../specification/searchSpecification.do?pageNum=" + currentPage
            + "&pageSize=" + itemsPerPage, searchEntity);
    }

    this.selectSpecificationVoById = function (id) {
        return $http.get("../specification/selectSpecificationVoById.do?id=" + id);
    }

    this.insertSpecificationVo = function (specificationVo) {
        return $http.post("../specification/insertSpecificationVo.do", specificationVo);
    }

    this.updateSpecificationVo = function (specificationVo) {
        return $http.post("../specification/updateSpecificationVo.do", specificationVo);
    }

    this.deleteSpecificationVo = function (ids) {
        return $http.post("../specification/deleteSpecificationVo.do",ids);
    }

    this.selectAllSpecificationMap = function () {
        return $http.get("../specification/selectAllSpecificationMap.do");
    }
});